play/square-wars/script.js

```diff
@@ -0,0 +1,715 @@
+    }
+  }
+
+  /**
+    * Method for handling crashes with bosses
+    * if the bullet hits the boss slice the enemy from the array and add a kill count
+    * does not give points
+    **/
+  checkCrashHandleBoss() {
+    for (let i = 0; i < bullets.length; i++) {
+      for (let j = 0; j < boss.length; j++) {
+        if (bullets[i].crashWith(boss[j]) && boss[j] != undefined) {
+          boss.splice(boss.indexOf(boss[j]), 1);
+        }
+      }
+    }
+  }
+
+  //spawn bullet from player shooting
+  spawnBullet() {
+    let bullet = new Bullet(5, 10, -2, -2, 'black');
+    bullet.shoot();
+    bullets.push(bullet);
+  }
+
+  /**
+  * spawn enemies method spawns enemies based on the round and interval
+  * fromula interval 120 (constant) - round % 1 == 0
+  * This allows for enemies to spawn based off of current round
+  **/
+  spawnEnemies() {
+    let randomX = Math.floor(Math.random() * 240);
+    if (myGameArea.framNo === 1 || myGameArea.everyInterval(120 - player.roundCount)) {
+      enemys.push(new Npc(30, 30, randomX, -5, 'blue'));
+
+      //spawn boss fomula round % 3 == 0 count is to hold how many are spawn
+      if (this.round % 3 == 0 && this.round != 0) {
+        //turn killcount to string to send data to compare
+        let holdData = player.roundCount.toString();
+        holdData = holdData.split('');
+        holdData = holdData.slice(0, 1);
+        parseInt(holdData);
+        this.spawnBoss(holdData)
+      }
+
+    }
+  }
+
+
+  //update enemies method to manage enemies speed on canvas
+  updateEnemies() {
+    //loop threw enemys and update
+    for (let i = 0; i < enemys.length; i++) {
+      //if player has killCount of 100 enemys move faster
+      if (player.killCount >= 100) {
+        enemys[i].y += 1.30;
+      } else {
+        enemys[i].y += 1.25;
+      }
+      enemys[i].update();
+    }
+  }
+
+  //update bullets to maintain speed and position on canvas
+  updateBullets() {
+    //update bullet
+    for (let j = 0; j < bullets.length; j++) {
+      if (bullets[j].y <= 0) {
+        bullets.splice(bullets.indexOf(bullets[j]), 1);
+      }
+      if(bullets[j] != undefined) {
+        bullets[j].newPos();
+        bullets[j].update();
+      }
+    }
+  }
+
+  //update boss to maintain speed and position on canvas
+  updateBoss() {
+    if(boss.length != 0) {
+      boss.forEach((el) => {
+        el.y += 1;
+        el.newPos();
+        el.update();
+      });
+    }
+  }
+
+  /**
+  * This is called from the game engine every 20millsec so having a count variable limits the amout of
+  * bosses. equation is: time(millsec), count(int) round(int) spawn(int):
+  * equation = while(round is a boss round aka round % 3 == 0 and count  < round) spawn = time*count
+  * this equation and method is only run depending on spawn enemies equation 120-playerkillCount % 1 === 0
+  * @param {number} count
+  **/
+  spawnBoss(count) {
+    let randomX = Math.floor(Math.random() * 240);
```

Comment:
The 240 should be a constant just in case you want to change it some day, but also to provide context and meaning to the number.